### PR TITLE
Fix formatting aliases

### PR DESCRIPTION
**Pull request by ra0x3**
### Description
- This PR fixes an issue where `forc-fmt` would not respect aliases and would turn `use foo::{bar as Bar};` into `use foo::{barasBar};`
  - The fix is done by swapping `.match_indices()` with a similar function `match_indices_str()` which can match the full `as` token

### Testing steps
- [ ] A test was added so `cargo test -p sway-fmt` should pass
- [ ] Create a new `forc init` project and try out some of the following use statements:

```rust
// This...
use a::b::{c as d, e as f};

// Formats to...
use a::b::{c as d, e as f};

// And this...
use std::{address::*, assert::assert as LocalAssert, block::*, chain::auth::*, context::{*,text::{call_frames::*, dial_frames::{Transaction as DialFrameTransaction, TransactionParameters}, token_storage::{CallData, Parameters}}}, contract_id::ContractId, hash::*, panic::panic, storage::*, token::*};

// Formats to...
use std::{
    address::*,
    assert::assert as LocalAssert,
    block::*,
    chain::auth::*,
    context::{
        *,
        text::{
            call_frames::*,
            dial_frames::{Transaction as DialFrameTransaction, TransactionParameters},
            token_storage::{CallData, Parameters}
        }
    },
    contract_id::ContractId,
    hash::*,
    panic::panic,
    storage::*,
    token::*,
};
```

### CODE REVIEW

**Review by adlerjohn — COMMENTED** [review]
lgtm, but will defer to @JoshuaBatty and @mitchmindtree

File: sway-fmt/src/traversal_helper.rs
Comment (mitchmindtree) [inline sway-fmt/src/traversal_helper.rs:99]: Can `s.len()` ever be less than `4`? If so, I think the `s.len() - 4` expression may underflow and `panic!` or wrap around as a result.

Comment (mitchmindtree) [reply sway-fmt/src/traversal_helper.rs:99]: Also, where does `4` come from? Can we get this from a const or retrieving the length directly from what this refers to?

File: sway-fmt/src/traversal_helper.rs
Comment (mitchmindtree) [inline sway-fmt/src/traversal_helper.rs:118]: Generally, it's a good idea to avoid iterating through `str`'s one `byte` at a time. This is because characters encoded in `UTF8` do not always have a consistent byte length.

This is a common source of confusion with Rust string handling, particularly w.r.t. use of `len`, indexing and slicing! The [`str::len`](https://doc.rust-lang.org/std/primitive.str.html#method.len) docs briefly demonstrate how the byte-length and character-length can differ.

One alternative approach for this loop might be to use:

```rust
for ch in s.chars() {
```

and then bump the `start` byte slice position with
```rust
    start += ch.len_utf8();
}
```

**Review by mitchmindtree — CHANGES_REQUESTED** [review]

File: sway-fmt/src/traversal_helper.rs
Comment (ra0x3) [reply sway-fmt/src/traversal_helper.rs:99]: I can make this change 👍

File: sway-fmt/src/traversal_helper.rs
Comment (ra0x3) [reply sway-fmt/src/traversal_helper.rs:118]: @mitchmindtree 
- I see what you mean here, and this is a good thing to keep in mind
- However, for this particular use case, don't `use` statements guarantee that our `str` is limited to characters from the set `abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789;}{,` ? which means that 1 char is guaranteed to be one byte?

Comment (adlerjohn) [reply sway-fmt/src/traversal_helper.rs:121]: Looks like identifiers can only be ASCII, so it might be fine to assume that. Morever, this will be replaced in the future with using the parse tree instead of manual string examination, so doesn't have to be completely perfect.

Comment (mitchmindtree) [reply sway-fmt/src/traversal_helper.rs:121]: > Morever, this will be replaced in the future with using the parse tree instead of manual string examination, so doesn't have to be completely perfect.

Yeah agreed, happy to leave this as is pending the new formatter.

> Looks like identifiers can only be ASCII, so it might be fine to assume that

Just to address this for future cases: Even if identifiers *should* only be ASCII in sway, it doesn't mean users won't try inputting non-ascii text and end up with LSP or formatter `panic!`ing. Any time we use `String` or `str` (everywhere), we make the implicit assumption that we're handling UTF8. Unless we validate that we're loading pure-ASCII when loading Sway files in `sway-core` itself and use ASCII string types everywhere to assure this at the type-level, we shouldn't make assumptions like this.

**Review by mitchmindtree — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

File: sway-fmt/src/traversal_helper.rs
Comment (ra0x3) [reply sway-fmt/src/traversal_helper.rs:121]: @mitchmindtree Thanks for that context. Even though this has been approved, let me see if I can update it to match our convention :)

Comment (adlerjohn) [reply sway-fmt/src/traversal_helper.rs:121]: @ra0x3 Maybe merge in as-is (since at least the current code _works_), then if there's anything to improve it can be done in a follow-up PR.

**Pull request merged**
